enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -340,13 +346,12 @@ def can_redeem(self, lms_user_id, content_key, skip_customer_user_check=False):
         )
         existing_transactions = subsidy_can_redeem_payload.get('all_transactions', [])
 
-        if not subsidy_can_redeem_payload.get('can_redeem', False):
-            return (False, REASON_NOT_ENOUGH_VALUE_IN_SUBSIDY, existing_transactions)
-
-        content_metadata = self.get_content_metadata(content_key)
         if not content_metadata:
             return (False, REASON_CONTENT_NOT_IN_CATALOG, existing_transactions)
 
+        if not subsidy_can_redeem_payload.get('can_redeem', False):
+            return (False, REASON_NOT_ENOUGH_VALUE_IN_SUBSIDY, existing_transactions)
+
```

Comment:
Logic swapped to avoid additional call into the subsidies `can_redeem`. Check if the `content_metadata` exist in the catalog before checking on a subsidy level.